dotCMS/src/main/java/com/dotcms/saml/DotSamlProxyFactory.java

```diff
@@ -0,0 +1,267 @@
+
+    private static final String ASSETS_PATH   = Config.getStringProperty("ASSET_REAL_PATH",
+            FileUtil.getRealPath(Config.getStringProperty("ASSET_PATH", "/assets")));
+    private static final String IDP_FILE_PATH = ASSETS_PATH + PROPERTIES_PATH;
+
+    private final MessageObserver    messageObserver    = new DotLoggerMessageObserver();
+    private final AppsAPI            appsAPI            = APILocator.getAppsAPI();
+    private final IdentityProviderConfigurationFactory identityProviderConfigurationFactory =
+            new DotIdentityProviderConfigurationFactoryImpl(this.appsAPI, APILocator.getHostAPI());
+
+    private SamlServiceBuilder        samlServiceBuilder;
+    private SamlConfigurationService  samlConfigurationService;
+    private SamlAuthenticationService samlAuthenticationService;
+
+    private static class SingletonHolder {
+
+        private static final DotSamlProxyFactory INSTANCE = new DotSamlProxyFactory();
+    }
+    /**
+     * Get the instance.
+     * @return DotSamlFactory
+     */
+    public static DotSamlProxyFactory getInstance() {
+
+        addRedirects();
+        return DotSamlProxyFactory.SingletonHolder.INSTANCE;
+    } // getInstance.
+
+    /**
+     * Key for the discard non-SAML AppSecretSavedEvent
+     * @return Comparable
+     */
+    @Override
+    public Comparable getKey() {
+        return SAML_APP_CONFIG_KEY;
+    }
+
+    /**
+     * When
+     * @param event
+     */
+    @Override
+    public void notify(final AppSecretSavedEvent event) {
+
+        final  Map<String, Secret> secretMap = event.getAppSecrets().getSecrets();
+        if (null != secretMap) {
+
+            SamlValidator.validateURL("sPEndpointHostname", secretMap.get("sPEndpointHostname").getString(), event.getUserId());
+            SamlValidator.validateXML("idPMetadataFile",    secretMap.get("idPMetadataFile").getString(),    event.getUserId());
+        }
+    }
+
+    /**
+     * Returns the dotCMS implementation of the identity provider config.
+     * This one basically returns (if exists) the configuration for a idp for a host
+     * @return IdentityProviderConfigurationFactory
+     */
+    public IdentityProviderConfigurationFactory identityProviderConfigurationFactory() {
+
+        return identityProviderConfigurationFactory;
+    }
+
+    private static void addRedirects() {
+
+        final NormalRule rule = new NormalRule();
+        rule.setFrom("^\\/dotsaml\\/("+String.join("|", DotSamlResource.dotsamlPathSegments)+")\\/(.+)$");
+        rule.setToType("forward");
+        rule.setTo("/api/v1/dotsaml/$1/$2");
+        rule.setName("Dotsaml REST Service Redirect");
+        DotUrlRewriteFilter urlRewriteFilter = DotUrlRewriteFilter.getUrlRewriteFilter();
+        try {
+            if(urlRewriteFilter != null) {
+                urlRewriteFilter.addRule(rule);
+            }else {
+                throw new Exception();
+            }
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)